src/v/cluster/rm_stm.cc

```diff
@@ -1827,6 +1829,26 @@ void rm_stm::apply_checkpoint(const model::record_batch& batch) {
     vlog(clusterlog.info, "Parked checkpoint state from term {}", term);
 }
 
+void rm_stm::apply_checkpoint_purge(const model::record_batch& batch) {
+    vassert(
+      batch.record_count() == 1, "Checkpoint batch with multiple records");
+
+    auto r = batch.copy_records();
+    auto& record = *r.begin();
+    auto key_buf = record.release_key();
+
+    auto term = serde::from_iobuf<model::term_id>(std::move(key_buf));
+
+    auto am_leader = term == _c->term() && _c->is_leader();
```

Comment:
don't need this check, the current leader may just take the _parked_checkpointed_mem_state before issuing the purge command so this logic will be simpler, just reset the parked state